x/incentives/keeper/iterator.go

```diff
@@ -63,11 +63,11 @@ func (k Keeper) FinishedGaugesIterator(ctx sdk.Context) sdk.Iterator {
 }
 
 // FilterLocksByMinDuration returns locks whose lock duration is greater than the provided minimum duration.
-func FilterLocksByMinDuration(locks []lockuptypes.PeriodLock, minDuration time.Duration) []lockuptypes.PeriodLock {
-	filteredLocks := make([]lockuptypes.PeriodLock, 0, len(locks))
+func FilterLocksByMinDuration(locks []lockuptypes.PeriodLock, minDuration time.Duration) []*lockuptypes.PeriodLock {
+	filteredLocks := make([]*lockuptypes.PeriodLock, 0, len(locks))
 	for _, lock := range locks {
 		if lock.Duration >= minDuration {
-			filteredLocks = append(filteredLocks, lock)
+			filteredLocks = append(filteredLocks, &lock)
```

Comment:
Does this work? I thought this would have every filtered lock point to the same loop variable, not the copies in the original slice. (Which would be &locks[i])

Reply:
It synched through epoch but I applied the suggestion and rerunning pprof

Reply:
Looks like my suggstion doesnt work by go tests failing

Reply:
Force pushed an update that I think should work. Need to reprofile it now as it leads to having to take a copy of the pointer in another loop

Reply:
With the bug fix:
![image](https://github.com/osmosis-labs/osmosis/assets/34196718/dddd164e-1e68-49b6-b3f5-8a379a2a9816)


Still much better overall. I will now do a long-lasting test as opposed to epoch only